src/vcpkg/configuration.cpp

```diff
@@ -535,6 +537,61 @@ namespace
 
 namespace vcpkg
 {
+    static ExpectedL<Optional<std::string>> get_baseline_from_git_repo(const VcpkgPaths& paths, StringView url)
+    {
+        auto res = paths.git_fetch_from_remote_registry(url, "HEAD");
+        if (auto p = res.get())
+        {
+            return Optional<std::string>(std::move(*p));
+        }
+        else
+        {
+            return msg::format(msgUpdateBaselineGitError, msg::url = url)
+                .appendnl()
+                .append_raw(Strings::trim(res.error()));
+        }
+    }
+
+    static constexpr StringLiteral BUILTIN_GIT_URL = "https://github.com/microsoft/vcpkg";
```

Comment:
This should be deduplicated against the string in src/vcpkg/registries.cpp:1249 (`vcpkg::make_builtin_registry()`)